tests/posix/c_lib_ext/src/getopt.c

```diff
@@ -55,38 +87,57 @@ enum getopt_idx {
 
 ZTEST(posix_c_lib_ext, test_getopt)
 {
-	struct getopt_state *state;
 	static const char *test_opts = "ac:";
 	static const char *const nargv[] = {
 		[GETOPT_IDX_CMD_NAME] = "cmd_name",
 		[GETOPT_IDX_OPTION1] = "-a",
 		[GETOPT_IDX_OPTION2] = "-c",
 		[GETOPT_IDX_OPTARG] = "foo",
+		NULL,
 	};
-	int argc = ARRAY_SIZE(nargv);
+	int argc = ARRAY_SIZE(nargv) - 1;
 	char **argv;
 	int c;
 
-	/* Get state of the current thread */
-	getopt_init();
-
 	argv = (char **)nargv;
+	optind = 0;
 
-	/* Test uknown option */
+	/* Test unknown option */
 
 	c = getopt(argc, argv, test_opts);
 	zassert_equal(c, 'a', "unexpected opt character");
 	c = getopt(argc, argv, test_opts);
 	zassert_equal(c, 'c', "unexpected opt character");
+	zassert_equal(0, strcmp(argv[GETOPT_IDX_OPTARG], optarg),
+		      "unexpected optarg result");
+}
 
-	c = getopt(argc, argv, test_opts);
-	state = getopt_state_get();
+ZTEST(posix_c_lib_ext, test_getopt_r)
+{
+	static const char *test_opts = "ac:";
+	static const char *const nargv[] = {
+		[GETOPT_IDX_CMD_NAME] = "cmd_name",
+		[GETOPT_IDX_OPTION1] = "-a",
+		[GETOPT_IDX_OPTION2] = "-c",
+		[GETOPT_IDX_OPTARG] = "foo",
+		NULL,
```

Comment:
Does not having a `NULL` cause something to fail?

Reply:
Yes, newlib and (current) picolibc crash.